packages/tenant-process/src/services/tenantService.ts

```diff
@@ -381,6 +385,70 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async addDeclaredAttribute({
+      tenantAttributeSeed,
+      authData,
+      correlationId,
+    }: {
+      tenantAttributeSeed: ApiDeclaredTenantAttributeSeed;
+      authData: AuthData;
+      correlationId: string;
+    }): Promise<Tenant> {
+      logger.info(
+        `Add declared attribute ${tenantAttributeSeed.id} to requester tenant ${authData.organizationId}`
+      );
+      const targetTenant = await retrieveTenant(
+        authData.organizationId,
+        readModelService
+      );
+
+      const maybeDeclaredTenantAttribute = targetTenant.data.attributes.find(
+        (attr) => attr.id === tenantAttributeSeed.id
+      );
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant: Tenant = {
+        ...targetTenant.data,
+        updatedAt: new Date(),
+      };
```

Comment:
@Carminepo2 do you have suggestions on how to avoid `let updatedTenant`? We need it to be assigned based on the logic below.

Reply:
My proposal:

```suggestion
      function assignDeclaredAttribute(): TenantAttribute[] {
        return [
          ...targetTenant.data.attributes,
          {
            id: unsafeBrandId(attribute.id),
            type: tenantAttributeType.DECLARED,
            assignmentTimestamp: new Date(),
            revocationTimestamp: undefined,
          },
        ];
      }

      function reassignDeclaredAttribute(): TenantAttribute[] {
        return targetTenant.data.attributes.map((attr) =>
          attr.id === attribute.id
            ? {
                ...attr,
                assignmentTimestamp: new Date(),
                revocationTimestamp: undefined,
              }
            : attr
        );
      }

      const updatedTenant: Tenant = {
        ...targetTenant.data,
        attributes: !maybeDeclaredTenantAttribute
          ? assignDeclaredAttribute()
          : reassignDeclaredAttribute(),
        updatedAt: new Date(),
      };
```

Reply:
I would only improve the naming of the two functinons, let me know if something better comes to your mind